libcloudforensics/providers/aws/internal/iam.py

```diff
@@ -49,125 +50,209 @@ def __init__(self,
+
+    Raises:
+      ResourceNotFoundError: If the profile failed creation due to already
+        existing, but then could not be found
     """
+    logger.info('Creating IAM Instance Profile {0:s}'.format(name))
     try:
-      logger.info("Creating IAM Instance Profile {0:s}".format(name))
       profile = self.client.create_instance_profile(InstanceProfileName=name)
-      return str(profile['InstanceProfile']['Arn'])
+      return str(profile['InstanceProfile']['Arn']), True
     except self.client.exceptions.EntityAlreadyExistsException as exception:
-      logger.info("Instance Profile exists already, using existing")
+      logger.info('Instance Profile exists already, using existing')
       profiles = self.client.list_instance_profiles()
 
       while True:
         for profile in profiles['InstanceProfiles']:
           if profile['InstanceProfileName'] == name:
-            return str(profile['Arn'])
+            return str(profile['Arn']), False
         if not profiles['IsTruncated']:
-          # If we reahed here it means the profile was deleted between the
+          # If we reached here it means the profile was deleted between the
           # creation failure and lookup
-          raise errors.ResourceNotFoundError('Could not locate instance profile with name {0:s} after creation failure due to EntityAlreadyExistsException' # pylint: disable=line-too-long
+          # pylint: disable=line-too-long
+          raise errors.ResourceNotFoundError('Could not locate instance profile with name {0:s} after creation failure due to EntityAlreadyExistsException'
+          # pylint: enable=line-too-long
             .format(name), __name__) from exception
 
-      profiles = self.client.list_instance_profiles(Marker=profiles['Marker'])
+        profiles = self.client.list_instance_profiles(Marker=profiles['Marker'])
+
+  def DeleteInstanceProfile(self, profile_name: str) -> None:
+    """Deletes an instance profile.
+
+    Args:
+      profile_name (str): The name of the instance profile to delete.
+    """
+    logger.info('Deleting instance profile {0:s}'.format(profile_name))
+    try:
+      self.client.delete_instance_profile(InstanceProfileName=profile_name)
+    except self.client.exceptions.NoSuchEntityException:
+      logger.info('IAM role {0:s} did not exist'.format(profile_name))
 
-  def CreateRole(self, name: str, assume_role_policy_doc: str) -> str:
+  def CreateRole(self, name: str, assume_role_policy_doc: str) \
+    -> Tuple[str, bool]:
     """Create an AWS IAM role. If it exists, return the existing.
 
     Args;
-      name: The name of the role.
-      assume_role_policy_doc: Assume Role policy doc.
+      name (str): The name of the role.
+      assume_role_policy_doc (str): Assume Role policy doc.
+
     Returns:
-      str: The Arn of the role.
+      Tuple[str, bool]: A tuple
+        str: The Arn of the role.
+        bool: True if the policy was created, false if it existed already.
```

Comment:
Same comments as before - I assume the returning boolean is if the Role was created and not the policy?